admin/Default/1.6/universe_create_warps.php

```diff
@@ -23,15 +23,14 @@
 }
 
 //get totals
-$db->query('SELECT * FROM warp WHERE game_id='.$db->escapeNumber($var['game_id']));
+$db->query('SELECT sector_id, warp FROM sector WHERE warp IS NOT NULL AND game_id='.$db->escapeNumber($var['game_id']));
 while ($db->nextRecord()) {
-	$warp1 = SmrSector::getSector($db->getInt('game_id'), $db->getInt('sector_id_1'));
-	$warp2 = SmrSector::getSector($db->getInt('game_id'), $db->getInt('sector_id_2'));
+	$warp1 = SmrSector::getSector($var['game_id'], $db->getInt('sector_id'));
+	$warp2 = SmrSector::getSector($var['game_id'], $db->getInt('warp'));
 	if ($warp1->getGalaxyID() == $warp2->getGalaxyID()) {
-		$warps[$warp1->getGalaxyID()][$warp2->getGalaxyID()]++;
+		$warps[$warp1->getGalaxyID()][$warp2->getGalaxyID()] += 0.5;
```

Comment:
I think you should add a comment as to why this is `+= .5` because it's not immediately obvious and having it explained in a comment could save a lot of time when reading the code

Reply:
Excellent point. I'll do this now.